service/src/main/java/bio/terra/landingzone/job/LandingZoneJobService.java

```diff
@@ -363,6 +365,42 @@ public void verifyUserAccess(BearerToken bearerToken, String jobId) {
     }
   }
 
+  public void verifyUserAccessForDeleteJobResult(
+      BearerToken bearerToken, UUID landingZoneId, String jobId) {
+    try {
+      FlightState flightState = stairwayComponent.get().getFlightState(jobId);
+      FlightMap inputParameters = flightState.getInputParameters();
+      UUID flightLandingZoneId =
+          inputParameters.get(LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+      if (!flightLandingZoneId.equals(landingZoneId)) {
+        throw new JobNotFoundException(
+            "The landing zone does not exist for job or the landing zone id is invalid.");
+      }
+
+      // If the flight is completed successfully only check if the user is a valid SAM user
```

Comment:
This basically removes all authorization once a landing zone is deleted. Not sure how big a hole that is, but it does leak some information about the LZ/billing-profile to users who don't have access.

I'm wondering if it would be more correct to check the user has access to the billing-profile when polling for LZ deletion. (Agree we shouldn't rely on the landing-zone Sam resource because it's being deleted :)).

Reply:
This is a valid concern and one that I struggled with as well - the billing profile is not leaked though, the job returns the ids of the deleted resources and lz id (which the user already knows). I considered using the billing profile. However, we can't get the billing profile from the landing zone id, but we plan to change the model to allow that. I was planning to add a ticket to change this once the model is changed.

Reply:
Makes sense -- maybe just add a TODO with the ticket to change this once the LZ model is updated?